Jolt/Core/Core.h

```diff
@@ -6,6 +6,8 @@
 // Determine platform
 #if defined(JPH_PLATFORM_BLUE)
 	// Correct define already defined, this overrides everything else
+#elif defined(__MINGW64__) || defined (__MINGW32__)
+    #define JPH_PLATFORM_MINGW
```

Comment:
Is MINGW a platform? It feels off among 'Windows', 'Linux', 'Android', 'iOS' etc. You can use the Win32 API from MINGW as far as I know, so to me it feels like it should have been JPH_COMPILER_MINGW.

Reply:
MinGW is not a compiler on its own. You can use clang ang gcc under MinGW. MinGW is not at OS,I would say it is in between the OS and the compiler.

MSYS2 has various "environments" which are different by the C and C++ library used: https://www.msys2.org/docs/environments/ (MSYSTEM environment variable).

So I thought the easiest way (in terms of quantity of code to change) to support it was simply to add a define for the platform. I only checked gcc and the mingw64 "environment" (I did not think about supporting the other ones).